### ISSUE TITLE
Logix nodes that cause exceptions should 'crash' aswell

### ISSUE THREAD

**Issue by mralext20** [Frooxius/NeosPublic#838]
Hello, recently, some broken code caused a 1.2 GB log file. 

Currently: Logix nodes that cause exceptions, when they cause exceptions, 'crash' the output, like so: https://this-is-a.stale-me.me/i/fvo1.png

i belive this should also 'crash' the offending node that causes the error.

reference exceptions from logs: 

```
Exception:
System.ArgumentOutOfRangeException: Count cannot be less than zero.
Parameter name: count
  at System.String.Remove (System.Int32 startIndex, System.Int32 count) [0x0002c] in <fb001e01371b4adca20013e0ac763896>:0 
  at FrooxEngine.LogiX.String.StringRemove.get_Content () [0x00053] in <f17f03b484854fecb3e5f1d76f043d33>:0 
  at FrooxEngine.LogiX.Input`1[T].EvaluateRaw (T def) [0x0000a] in <f17f03b484854fecb3e5f1d76f043d33>:0 
  at FrooxEngine.LogiX.Input`1[T].Evaluate (T def) [0x00010] in <f17f03b484854fecb3e5f1d76f043d33>:0 
  at FrooxEngine.LogiX.Display.Display_String.OnChanges () [0x00000] in <f17f03b484854fecb3e5f1d76f043d33>:0 
  at FrooxEngine.ComponentBase`1[C].InternalRunApplyChanges (System.Int32 updateIndex) [0x0000e] in <f17f03b484854fecb3e5f1d76f043d33>:0 
  at FrooxEngine.UpdateManager.ProcessChange (FrooxEngine.IUpdatable updatable) [0x00024] in <f17f03b484854fecb3e5f1d76f043d33>:0 
  at FrooxEngine.UpdateManager.RunQueue[T] (System.Collections.Generic.Queue`1[T] queue, System.Action`1[T] action) [0x00009] in <f17f03b484854fecb3e5f1d76f043d33>:0 
3:01:31 AM.974 (242 FPS)	Exception when Updating object: Element: ID1A01900, Type: FrooxEngine.LogiX.Display.Display_String, World: alex from alaska's Home, IsRemoved: False, IsDestroyed: False, IsDisposed: False, Enabled: False
Element: ID19E9B00, Type: FrooxEngine.Slot, World: alex from alaska's Home, IsRemoved: False, Slot name: Display, T: [0.2841675; 1.491582; 5.041621], R: [-0.129906; 0.4342507; 0.0634471; 0.8891149], S: [1; 1; 1], ActiveSelf: True, IsDestroyed: False
Element: ID1F00, Type: FrooxEngine.Slot, World: alex from alaska's Home, IsRemoved: False, Slot name: Root, T: [0; 0; 0], R: [0; 0; 0; 1], S: [1; 1; 1], ActiveSelf: True, IsDestroyed: False
Element: ID0, Type: FrooxEngine.World, World: alex from alaska's Home, IsRemoved: False



Exception:
System.ArgumentOutOfRangeException: Count cannot be less than zero.
Parameter name: count
  at System.String.Remove (System.Int32 startIndex, System.Int32 count) [0x0002c] in <fb001e01371b4adca20013e0ac763896>:0 
  at FrooxEngine.LogiX.String.StringRemove.get_Content () [0x00053] in <f17f03b484854fecb3e5f1d76f043d33>:0 
  at FrooxEngine.LogiX.Input`1[T].EvaluateRaw (T def) [0x0000a] in <f17f03b484854fecb3e5f1d76f043d33>:0 
  at FrooxEngine.LogiX.Input`1[T].Evaluate (T def) [0x00010] in <f17f03b484854fecb3e5f1d76f043d33>:0 
  at FrooxEngine.LogiX.Display.Display_String.OnChanges () [0x00000] in <f17f03b484854fecb3e5f1d76f043d33>:0 
  at FrooxEngine.ComponentBase`1[C].InternalRunApplyChanges (System.Int32 updateIndex) [0x0000e] in <f17f03b484854fecb3e5f1d76f043d33>:0 
  at FrooxEngine.UpdateManager.ProcessChange (FrooxEngine.IUpdatable updatable) [0x00024] in <f17f03b484854fecb3e5f1d76f043d33>:0 
  at FrooxEngine.UpdateManager.RunQueue[T] (System.Collections.Generic.Queue`1[T] queue, System.Action`1[T] action) [0x00009] in <f17f03b484854fecb3e5f1d76f043d33>:0 
2:59:07 AM.883 (347 FPS)	Exception when Updating object: Element: ID18DBD00, Type: FrooxEngine.LogiX.Display.Display_String, World: alex from alaska's Home, IsRemoved: False, IsDestroyed: False, IsDisposed: False, Enabled: False
Element: ID18C3F00, Type: FrooxEngine.Slot, World: alex from alaska's Home, IsRemoved: False, Slot name: Display, T: [0.2826532; 1.453604; 5.059814], R: [-0.1038259; 0.425228; 0.04904766; 0.8977727], S: [1; 0.9999999; 1], ActiveSelf: True, IsDestroyed: False
Element: ID1F00, Type: FrooxEngine.Slot, World: alex from alaska's Home, IsRemoved: False, Slot name: Root, T: [0; 0; 0], R: [0; 0; 0; 1], S: [1; 1; 1], ActiveSelf: True, IsDestroyed: False
Element: ID0, Type: FrooxEngine.World, World: alex from alaska's Home, IsRemoved: False


```

**Comment by Frooxius** [comment]
Unfortunately this is not possible without adding a significant overhead and performance impact. Generally exceptions are unexpected situations, that aren't handled correctly, so you will get unpredictable behavior. The main problem here is that the node is throwing the exception in the first place, which it should never do, no matter what inputs you give it.

I'll fix the bug in the particular node. If you find more nodes breaking and throwing an exceptions with certain inputs, please report those as well!

**Comment by Frooxius** [comment]
Fixed bug in the StringRemove node in 2020.8.3.1234! If you find any more please let me know. The "crashing" of all nodes won't be implemented due to performance considerations.
